include/mpi.h

```diff
@@ -126,22 +130,23 @@ typedef struct MPI_ABI_Datatype * MPI_Datatype;
 #define MPI_UINT32_T                   (MPI_Datatype)0x00000251
 #define MPI_INT64_T                    (MPI_Datatype)0x00000258
 #define MPI_UINT64_T                   (MPI_Datatype)0x00000259
-#define MPI_INTEGER1                   (MPI_Datatype)0x000002c0
-#define MPIX_LOGICAL1                  (MPI_Datatype)0x000002c1
+#define MPIX_LOGICAL1                  (MPI_Datatype)0x000002c0
+#define MPI_INTEGER1                   (MPI_Datatype)0x000002c1
 #define MPIX_REAL1                     (MPI_Datatype)0x000002c2
 #define MPI_CHARACTER                  (MPI_Datatype)0x000002c3
-#define MPI_INTEGER2                   (MPI_Datatype)0x000002c8
-#define MPIX_LOGICAL2                  (MPI_Datatype)0x000002c9
-#define MPIX_REAL2                     (MPI_Datatype)0x000002ca
-#define MPI_INTEGER4                   (MPI_Datatype)0x000002d0
-#define MPIX_LOGICAL4                  (MPI_Datatype)0x000002d1
+#define MPIX_LOGICAL2                  (MPI_Datatype)0x000002c8
+#define MPI_INTEGER2                   (MPI_Datatype)0x000002c9
+#define MPI_REAL2                      (MPI_Datatype)0x000002ca
+#define MPIX_LOGICAL4                  (MPI_Datatype)0x000002d0
+#define MPI_INTEGER4                   (MPI_Datatype)0x000002d1
 #define MPI_REAL4                      (MPI_Datatype)0x000002d2
-#define MPIX_COMPLEX4                  (MPI_Datatype)0x000002d3
-#define MPI_INTEGER8                   (MPI_Datatype)0x000002d8
-#define MPIX_LOGICAL8                  (MPI_Datatype)0x000002d9
+#define MPI_COMPLEX4                   (MPI_Datatype)0x000002d3
+#define MPIX_LOGICAL8                  (MPI_Datatype)0x000002d8
+#define MPI_INTEGER8                   (MPI_Datatype)0x000002d9
 #define MPI_REAL8                      (MPI_Datatype)0x000002da
 #define MPI_COMPLEX8                   (MPI_Datatype)0x000002db
-#define MPI_INTEGER16                  (MPI_Datatype)0x000002e0
+#define MPIX_LOGICAL16                 (MPI_Datatype)0x000002e0
+#define MPI_INTEGER16                  (MPI_Datatype)0x000002e1
```

Comment:
What is the reason for these datatype constants swapping?

Reply:
Lisandro wants logical to come before integer in the Fortran datatype constants.  I don't care either way.

Reply:
This is bikeshedding from my side, but in my little brain, "logical" is narrower than "integer" so makes sense to list it first. If there is any counterargument,  I'm happy to revert this swapping. We still have time to revisit the actual values of all these constants.

Reply:
It might be nicer if we split critical changes and gratuitous changes into separate commits. I don't mind subjective/cosmetic changes, but if we are applying some implicit rules, then it is better to make it explicit via comments or commit messages.